src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -44,12 +59,408 @@
+    }
+
+    /**
+     *  Return an array with row sums in each entry
+     *
+     * @param matrix Never {@code null}
+     * @return Never {@code null}
+     */
+    public static double[] rowSums(final RealMatrix matrix) {
+        Utils.nonNull(matrix);
+
+        return IntStream.range(0, matrix.getRowDimension())
+                .mapToDouble(r -> sum(matrix.getRow(r))).toArray();
+    }
+
+    /**
+     *  Return sum of 3d array
+     *
+     * @param array Never {@code null}
+     * @return sum of array
+     */
+    public static double sum(final double[][][] array) {
+        Utils.nonNull(array);
+        double result = 0;
+        for (double[][] d: array) {
+            for (double[] da: d){
+                for (double daa: da) {
+                    result += daa;
+                }
+            }
+
+        }
+        return result;
+    }
+
+    public static int minIndex(final int ... values) {
+        Utils.nonNull(values);
+        if (values.length == 0) {
+            return -1;
+        }
+        int minValue = values[0];
+        int minIndex = 0;
+        for (int i = 0; i < values.length; i++) {
+            final int nextValue = values[i];
+            if (nextValue < minValue) {
+                minValue = nextValue;
+                minIndex = i;
+            }
+        }
+        return minIndex;
+    }
+
+    /**
+     * Given an int array returns the difference between the second smallest element
+     * and the smallest element.
+     * @param values the input values.
+     * @param defaultValue value to return in case the input array has zero or only one element.
+     * @return 0 if the input array has less than 2 elements, otherwise the second smallest
+     * - the smallest.
+     * @throws IllegalArgumentException if {@code values} is {@code null}.
+     */
+    public static int secondSmallestMinusSmallest(final int[] values, final int defaultValue) {
+        Utils.nonNull(values);
+        if (values.length <= 1) {
+            return defaultValue;
+        } else {
+            int smallest = values[0];
+            int secondSmallest = Integer.MAX_VALUE;
+            for (int i = 1; i < values.length; i++) {
+                if (values[i] < smallest) {
+                    secondSmallest = smallest;
+                    smallest = values[i];
+                } else if (values[i] < secondSmallest) {
+                    secondSmallest = values[i];
+                }
+            }
+            return secondSmallest - smallest;
+        }
+    }
+
+    /**
+     * Returns the smallest power of 2 that exceeds or equals {@code val}
+     * @param val input value
+     * @return power of 2 integer
+     * @throws IllegalArgumentException if an int overlow is encountered
+     */
+    public static int smallestPowerOfTwoGreaterThan(final int val)
+            throws IllegalArgumentException {
+        if (val > Integer.MAX_VALUE/2 || val < 0)
```

Comment:
Use Utils.validateArg, clean up typos, and remove `throws` from signature.

Reply:
done done done